src/ratelimit_action.rs

```diff
@@ -208,7 +207,14 @@ impl RateLimitAction {
                             };
                             entries.push(entry);
                         }
-                    });
+                        Err(e) if e.is_transient() => {
+                            //only interested in returning transient errors for so we can evaluate again later
+                            info!("build_message(rl): transient evaluation error for requestData field '{}': {}", field, e);
+                            return Err(BuildMessageError::Evaluation(Box::new(e)));
+                        }
+                        _ => {}
+                    }
+                }
```

Comment:
As far as I understand this should not be done and is not safe - unless something changed, you cannot delay the request until the requestBody phase as it'll stream the headers to the upstream cc @alexsnaps 

See this document on stop iteration: https://docs.google.com/document/d/1Whd1C0k-H2NHrPOmlAqqauFz6ObSTP017juJIYyciB0/edit?tab=t.0#heading=h.17wg41voij6q

Reply:
Thats' why we pre-fetch the request.* attributes in advance.

Reply:
Now I undestand your initial statement about "you cannot delay the request until the requestBody phase as it'll already have sent the headers to the upstream". If that would be true, we would not be able to rate limit based on anything from the request body. Right? At the very least, we know for sure that request body will not be routed to the upstream service.